firedrake/slate/slate.py

```diff
@@ -0,0 +1,767 @@
+"""SLATE is a symbolic language defining a framework for
```

Comment:
I definitely agree. I think perhaps `test_slate_hybridization.py` could be the template for constructing a full-demo. As for documentation, you mean adding documentation on SLATE that shows up in http://www.firedrakeproject.org/documentation.html?